dts/bindings/serial/uart-controller.yaml

```diff
@@ -23,3 +23,22 @@ properties:
       - "none"
       - "odd"
       - "even"
+  stop-bits:
+    type: string
+    description: |
+      Sets the number of stop bits. Default to 1 if not specified.
+    enum:
+      - "0_5"
+      - "1"
+      - "1_5"
+      - "2"
+  data-bits:
+    type: string
+    description: |
+      Sets the number of data bits. Default to 8 if not specified.
```

Comment:
- this doesn't justify defaults (required).
- seems to deviate from https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/serial/serial.yaml cc: @galak 
- how is this going to be used? app needs to call uart_configure() where those are passed.

Reply:
> how is this going to be used? app needs to call uart_configure() where those are passed.

Through normal driver initialisation? E.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_rpi_pico.c#L145 instead of these being static values, they can be retrieved from the devicetree for a UART instance.

Reply:
If we allow initial config from DT, then I guess we need to use the `zephyr,` prefix here. Then, this is something that should be handled by the driver class itself, to avoid repeating the same code on every single driver. See e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/regulator/regulator_common.c#L16 (still not ideal, our device model isn't great to do these things nicely).

Reply:
I understand it doesn't justify defaults, because if there is no implementation behind it (like for none of the drivers), then they will require runtime configuration? There probably is no good default to choose because it depends on the arch and platform? Then why is there a mention of default on parity?

Reply:
Removed mention of defaults.